bot.py

```diff
@@ -90,6 +90,30 @@ async def begin(ctx: commands.Context): # noqa: D103
 	await ctx.send("Profile created! `p!help` for more commands.")
 
 
+@bot.command(help='Add name to display specific Pokemon')
+async def show(ctx: commands.Context, single: Optional[str]): # noqa: D103
+	discord_id = ctx.author.id
+	base_msg = f"<@!{discord_id}> Here are all of your current Pokemon"
+	msg: Message = await ctx.send(base_msg)
+	#loads user class based upon their discord_id
+	user = await userprofile.load_profile(discord_id)
+	pokemon_list = [p async for p in user.pokemon_iter()]
```

Comment:
No need to do this, because if the user only wants a single pokemon, this forces all the pokemon the user owns to get loaded from the database.

`pokemon_iter()` gives you an iterator that you can iterate over. You're just iterating over `pokemon_list` later, so instead of 

```python
for pokemon in pokemon_list:
```
try
```python
async for pokemon in user.pokemon_iter():
```